### PR TITLE
ESQL: Load stored fields sequentially

### PR DESCRIPTION
**Pull request by nik9000**
The lucene APIs can load stored fields more quickly if flip them into "sequential" mode - but it's only faster when you are loading a dense set of documents - all bunched up. This flips the stored field loading code to use a sequential reader when the documents are fully dense - literally just counting without gaps - which is precisely what the fetch phase does.

### CODE REVIEW

**Comment by nik9000** [comment]
The tests I've been using only load a small number of documents so they don't see any benefit from this. I'm going to run a larger test set and see if that shows anything.

**Comment by nik9000** [comment]
Oh boy does this help if you run a `STATS` on a stored field:
```
before: 193984ms
 after:  82688ms
```

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java [hunk 1]
```diff
@@ -241,9 +248,24 @@ private void loadFromManyLeaves(Block[] blocks, DocVector docVector) throws IOEx
         }
     }
 
-    private void trackStoredFields(StoredFieldsSpec spec) {
+    /**
+     * Is it more efficient to use a sequential stored field reader
+     * when reading stored fields for the documents contained in {@code docIds}.
+     */
+    private boolean isSequential(IntVector docIds) {
+        return docIds.getInt(docIds.getPositionCount() - 1) - docIds.getInt(0) == docIds.getPositionCount() - 1;
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java:256]: I think we should only switch to the sequential mode when the number of documents exceeds a certain threshold? The search API uses 10 for this. The sequential (i.e., merging) stored field reader eagerly decompresses an entire block.

Comment (nik9000) [reply x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java:256]: I can do that! FWIW, we're not going to have many blocks less than 10, but it's fine to do.

**Comment by nik9000** [comment]
@dnhatn , I think this is ready for another round.

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java [hunk 2]
```diff
@@ -51,6 +51,18 @@
  * and outputs them to a new column.
  */
 public class ValuesSourceReaderOperator extends AbstractPageMappingOperator {
+    /**
+     * Minimum number of documents for which it is more efficient to use a
+     * sequential stored field reader when reading stored fields.
+     * <p>
+     *     The sequential stored field reader decompresses a whole block of docs
+     *     at a time so for very short lists it won't be faster to use it. We use
+     *     {@code 10} documents as the boundary for "very short" because it's what
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java:60]: 💯

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @nik9000

**Comment by nik9000** [comment]
Thanks @dnhatn !

**Pull request merged**